arq/connections.py

```diff
@@ -101,6 +108,8 @@ async def enqueue_job(
         expires_ms = to_ms(_expires)
 
         with await self as conn:
+            if _del_prev_result:
+                await conn.delete(result_key_prefix + job_id)
```

Comment:
why can't this be part of the pipelined operations?

Reply:
why is this needed as part of this PR.

I think you may need to separate your work here into multiple pull requests for easier reveiwing.